packages/eslint-plugin/src/rules/confusing-non-null-assertion-like-not-equal.ts

```diff
@@ -0,0 +1,80 @@
+import {
+  AST_NODE_TYPES,
+  AST_TOKEN_TYPES,
+  TSESLint,
+  TSESTree,
+} from '@typescript-eslint/experimental-utils';
+import * as util from '../util';
+
+type MessageIds = 'notNeedInEqualTest' | 'confusing';
+
+export default util.createRule<[], MessageIds>({
+  name: 'confusing-non-null-assertion-like-not-equal',
```

Comment:
I think a better name for the rule would be `no-confusing-non-null-assertion`.
There won't ever be an inverse of this rule, and this keeps us open in case we want to add more cases in the future.

Reply:
Yeah